### PR TITLE
WIP NEW Adding a field that can be used to select a CKAN resource from an API

### PR DESCRIPTION
**Pull request by ScopeyNZ**
- Adds a simple isomorphic-fetch wrapper for querying a CKAN api
- Adds a bare-bones PHP implementation of a field for selecting a resource from a CKAN API
- Adds a React component used to render the field

This is pretty much fully implemented now and ready for review. I'm just going through the process of manually testing and adding tests.

### CODE REVIEW

File: client/src/components/CKANResourceLocator.js [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    if (spec.resource) {
+      if (!spec.dataset) {
+        // If we have a resource but no dataset then we can determine the dataset from the resource
+        // response provided by the API.
+        CKANApi.loadResource(spec.endpoint, spec.resource).then(
+          resource => {
+            // Create a new spec with the updated dataset
+            const newSpec = {
+              ...spec,
+              dataset: resource.package_id,
```
[the remark this answers is not in the document]
Comment (NightJar) [reply client/src/components/CKANResourceLocator.js]: It turned out `package_name` was not available in the data returned from the api endpoint.

**Comment by NightJar** [comment]
I agree, but no. Follow up PR for PHP tests please.

File: src/Forms/ResourceLocatorField.php [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        $schemaData['defaultEndpoint'] = $this->getDefaultEndpoint();
+
+        return $schemaData;
+    }
+
+    public function setValue($value, $data = null)
+    {
+        // $value should be the child DataObject but if it isn't we can use `getSaveTarget`
+        if (!$value instanceof DataObjectInterface) {
+            $value = $this->getSaveTarget($data);
```
Comment (robbieaverill) [inline src/Forms/ResourceLocatorField.php]: There's a bug here - $data is nullable AND it unlikely to be a DataObjectInterface, but getSaveTarget has that interface type hinted

Comment (ScopeyNZ) [reply src/Forms/ResourceLocatorField.php]: Well spotted. Yeah we should assert that `$data instanceof DataObjectInterface` too. I'll follow up as part of the work I do with #18.

**Pull request merged**